### PR TITLE
Move to rails 6.1 defaults

### PR DESCRIPTION
**Pull request by SamJamCul**
## Context

https://trello.com/c/JlRb4Rr2/4965-set-rails-7-defaults

This moves to the rails 6.1 defaults. Only one test broke as a result, which was based on the `has_many_inversing` setting. The service that relied on this has been modified to avoid the issue.

### CODE REVIEW

File: app/services/save_provider_user.rb [hunk 1]
```diff
@@ -61,7 +58,7 @@ def send_permissions_granted_email
   end
 
   def new_provider_permissions
-    @new_provider_permissions ||= provider_permissions - provider_user.provider_permissions
+    @new_provider_permissions ||= provider_permissions - @old_provider_permissions
```
Comment (cwrw) [inline app/services/save_provider_user.rb:61]: can you add `old_provider_permissions` to `attr_reader` so we don't need to use `@`?

File: app/services/save_provider_user.rb [hunk 2]
```diff
@@ -4,13 +4,14 @@ class SaveProviderUser
   def initialize(provider_user:, provider_permissions: [])
     @provider_user = provider_user
     @provider_permissions = provider_permissions
+    @old_provider_permissions = provider_user.provider_permissions.select(&:persisted?)
```
Comment (cwrw) [inline app/services/save_provider_user.rb:7]:
```suggestion
    @old_provider_permissions = provider_user&.provider_permissions.select(&:persisted?)
```
for the failures

File: app/services/save_provider_user.rb (hunk 1 above)
Comment (SamJamCul) [reply app/services/save_provider_user.rb]: sorted

File: app/services/save_provider_user.rb (hunk 2 above)
Comment (SamJamCul) [reply app/services/save_provider_user.rb]: whoops, done now!

File: config/initializers/new_framework_defaults_6_1.rb [hunk 3]
```diff
@@ -1,67 +0,0 @@
...
-# Rails.application.config.active_storage.queues.purge = nil
-
-# Set the default queue name for the incineration job to the queue adapter default.
-# Rails.application.config.action_mailbox.queues.incineration = nil
-
-# Set the default queue name for the routing job to the queue adapter default.
-# Rails.application.config.action_mailbox.queues.routing = nil
-
-# Set the default queue name for the mail deliver job to the queue adapter default.
-# Rails.application.config.action_mailer.deliver_later_queue_name = nil
```
Comment (cwrw) [inline config/initializers/new_framework_defaults_6_1.rb:63]: I think we'll need to set this to the previous value `:mailers`

Comment (SamJamCul) [reply config/initializers/new_framework_defaults_6_1.rb:63]: right, so making sure it's explicitly set!

**Review by cwrw — APPROVED** [review]

**Pull request merged**